pkgs/os-specific/serenity/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, stdenv
+, stdenvNoCC
+, runCommand
+, callPackage
+, buildPackages
+, pkgsBuildBuild
+, patchelf
+}:
+let
+  rev = "0bbd9040efbe97850a18a49a9cea25498d727f13";
+
+  version = "unstable-2023-05-18";
+
+  # Fetch Serenity source and patch shebangs
+  src = buildPackages.stdenvNoCC.mkDerivation {
+    pname = "serenity-src";
+    inherit version;
+
+    src = buildPackages.fetchFromGitHub {
+      owner = "SerenityOS";
+      repo = "serenity";
+      sha256 = "ene8HLOtch4uGZ4nv61woDpRegRm9eLUl3n4V5jt0tA=";
+      inherit rev;
+    };
+
+    installPhase = ''
+      cp -r . $out
+    '';
+
+    dontPatchELF = true;
+  };
+
+  rootfs = callPackage ./rootfs.nix { inherit stdenv src version; };
+
+  serenity-headers = callPackage ./headers.nix { inherit src version; };
+
+  mkLibrary = pname: deps:
+    stdenvNoCC.mkDerivation {
+      inherit pname version;
+
+      # FIXME: rpath is not rewritten to use dependency derivations
+      buildInputs = builtins.attrValues deps;
```

Comment:
This is supposed to drop direct references to `serenity-rootfs` and replace them with dependencies defined in  `libraries.nix`. I've tried combinations of `patchelf` and `autoPatchelfHook` but without success